internal/services/network/application_gateway_resource.go

```diff
@@ -2607,6 +2940,74 @@ func flattenApplicationGatewayBackendHTTPSettings(input *[]network.ApplicationGa
 	return results, nil
 }
 
+func flattenApplicationGatewayBackendSettings(input *[]network.ApplicationGatewayBackendSettings) ([]interface{}, error) {
+	results := make([]interface{}, 0)
+	if input == nil {
+		return results, nil
+	}
+
+	for _, v := range *input {
+		output := map[string]interface{}{}
+
+		if v.ID != nil {
+			output["id"] = *v.ID
```

Comment:
Can we parse and then construct the ID, setting its `.ID()` value back to state here?